src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,290 @@
+    sas_th_url = config['url']
+    device_a = config['device_a']
+    device_b = config['device_b']
+    device_c = config['device_c']
+    device_d = config['device_d']
+    grant_a = config['grant_a']
+    grant_b = config['grant_b']
+    grant_c = config['grant_c']
+    grant_d = config['grant_d']
+    ppa_a = config['ppa_a']
+    esc_sensor = config['esc_sensor']
+    # Start the server
+    self._sas_server.startServer()
+
+    # Step 3: Whitelist the FCCID and UserID for the device_b in SAS UUT
+    self._sas_admin.InjectFccId({'fccId': device_b['fccId']})
+    self._sas_admin.InjectUserId({'userId': device_b['userId']})
+
+    # Step 4: load the device_b with registration to SAS UUT
+    request = {'registrationRequest': [device_b]}
+    response = self._sas.Registration(request)['registrationResponse'][0]
+    cbsd_id_b = response['cbsdId']
+    # Check registration response of device_b
+    self.assertEqual(response['response']['responseCode'], 0)
+    del request, response
+    # Step 5: Load the device_b with grant to SAS UUT
+    request = {'grantRequest': [grant_b]}
+    response = self._sas.Grant(request)['grantResponse'][0]
+    grant_id_b = response['grantId']
+    # Check grant response of device_b
+    self.assertEqual(response['response']['responseCode'], 0)
+    del request, response
+    # Whitelist the FCCID and UserID for the device_d in SAS UUT
+    self._sas_admin.InjectFccId({'fccId': device_d['fccId']})
+    self._sas_admin.InjectUserId({'userId': device_d['userId']})
+    # Step 8: Load the device_d with registration to SAS UUT
+    request = {'registrationRequest': [device_d]}
+    response = self._sas.Registration(request)['registrationResponse'][0]
+    cbsd_id_d = response['cbsdId']
+    # Check registration response of device_d of SAS UUT
+    self.assertEqual(response['response']['responseCode'], 0)
+    del request, response
+    # Step 9: Load the device_d with grant to SAS UUT
+    request = {'grantRequest': [grant_d]}
+    response = self._sas.Grant(request)['grantResponse'][0]
+    grant_id_d = response['grantId']
+    # Check grant response of device_d to SAS UUT
+    self.assertEqual(response['response']['responseCode'], 0)
+    del request, response
+    # Step 10: Send the Heatbeat request for the device_b to SAS UUT
+    request = {
+      'heartbeatRequest': [{
+        'cbsdId': cbsd_id_b,
+        'grantId': grant_id_b,
+        'operationState': 'GRANTED'
+      }]
+    }
+    response = self._sas.Heartbeat(request)['heartbeatResponse'][0]
+    self.assertEqual(response['response']['responseCode'], 0)
+    del request, response
+    # Step 10: Send the Heatbeat request for the device_d to SAS UUT
+    request = {
+      'heartbeatRequest': [{
+        'cbsdId': cbsd_id_d,
+        'grantId': grant_id_d,
+        'operationState': 'GRANTED'
+      }]
+    }
+    response = self._sas.Heartbeat(request)['heartbeatResponse'][0]
+    self.assertEqual(response['response']['responseCode'], 0)
```

Comment:
isn't it better to do one HB request with both grants?

Reply:
will change to do HB for both grants